src/Microsoft.TestPlatform.Execution.Shared/UILanguageOverride.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Globalization;
+
+namespace Microsoft.VisualStudio.TestPlatform.Execution
+{
+    internal static class UILanguageOverride
+    {
+        private const string DOTNET_CLI_UI_LANGUAGE = nameof(DOTNET_CLI_UI_LANGUAGE);
+        private const string VSLANG = nameof(VSLANG);
+        private const string PreferredUILang = nameof(PreferredUILang);
+
+        internal static void SetCultureSpecifiedByUser()
+        {
+            CultureInfo language = GetOverriddenUILanguage();
+            if (language == null)
+            {
+                return;
+            }
+
+            ApplyOverrideToCurrentProcess(language);
+            FlowOverrideToChildProcesses(language);
+        }
+
+
+        private static void ApplyOverrideToCurrentProcess(CultureInfo language)
+        {
+            CultureInfo.DefaultThreadCurrentUICulture = language;
+        }
+
+        private static CultureInfo GetOverriddenUILanguage()
+        {
+            // DOTNET_CLI_UI_LANGUAGE=<culture name> is the main way for users to customize the CLI's UI language.
+            string dotnetCliLanguage = Environment.GetEnvironmentVariable(DOTNET_CLI_UI_LANGUAGE);
```

Comment:
@nohwnd : Looking at the env variable name, just wanted to confirm that this works for .NET Framework and VS scenarios as well? Sounds like it does from your offline responses but thought I'll put it out here.

Reply:
Never mind, looks like that is covered by the code below.